package/MDAnalysis/analysis/dssp/dssp.py

```diff
@@ -0,0 +1,364 @@
+
+:Author: Egor Marin
+:Year: 2024
+:Copyright: LGPL v2.1+
+
+.. versionadded:: 2.8.0
+
+The module contains code to build hydrogend bond contact map,
+and use it to assign protein secondary structure (:class:`DSSP`).
+
+This module uses the python version of the original algorithm :footcite:p:`Kabsch1983`,
+re-implemented by @ShintaroMinami and available under MIT license from
+`ShintaroMinami/PyDSSP <https://github.com/ShintaroMinami/PyDSSP/tree/master#differences-from-the-original-dssp>`_.
+
+Note that this implementation does not discriminate different types of
+beta-sheets, as well as different types of helices, meaning you will get
+:math:`3_{10}` helices and π-helices labelled as "helix" too.
+
+
+Using original `pydssp`
+-----------------------
+
+The default implementation uses the original `pydssp` (v.0.9.0) code, rewritten
+without usage of `einops` library and hence having no dependencies. If you want
+to explicitly use `pydssp` (or its particular version), install it to your
+current environment with `python -m pip install pydssp`. Please note that the
+way MDAnalysis uses pydssp does not support pydssp's capability for batch
+processing or its use of the pytorch library.
+
+When using this module in published work please cite :footcite:p:`Kabsch1983`.
+
+.. rubric:: References
+
+.. footbibliography::
+
+
+Example applications
+--------------------
+
+Assigning secondary structure of a PDB file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example we will simply print a string representing a protein's secondary
+structure.
+
+.. code-block:: python
+
+    from MDAnalysis.tests.datafiles import PDB
+    from MDAnalysis.analysis.dssp import DSSP
+    u = mda.Universe(PDB)
+    s = ''.join(DSSP(u).run().results.dssp[0])
+    print(s)
+
+
+Calculating average secondary structure of a trajectory
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here we take a trajectory and calculate its average secondary structure, i.e.
+assign a secondary structure label 'X' to a residue if most of the frames in the
+trajectory got assigned the 'X' label.
+
+.. code-block:: python
+
+    from MDAnalysis.analysis.dssp import DSSP, translate
+    from MDAnalysisTests.datafiles import TPR, XTC
+    u = mda.Universe(TPR, XTC)
+    long_run = DSSP(u).run()
+    mean_secondary_structure = translate(long_run.results.dssp_ndarray.mean(axis=0))
+    print(''.join(mean_secondary_structure)[:20])
+
+Running this code produces ::
+
+   '-EEEEEE------HHHHHHH'
+
+Find parts of the protein that maintain their secondary structure during simulation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In this example, we will find residue groups that maintain their secondary structure
+along the simulation, and have some meaningful ('E' or 'H') secondary structure
+during more than set `threshold` share of frames. We will call these residues
```

Comment:
```suggestion
during more than set `threshold` fraction of the total frames. We will call these residues
```
Feel free to reject, it's just a nitpick - reading this I wasn't sure if the code below was going to expect a set number of frames or a fraction of the total number of frames.

Reply:
Indeed, thanks.